Add gRPC supported platforms doc (#21276)
dotnet/AspNetCore.Docs: aspnetcore/grpc/supported-platforms.md

---
title: gRPC for .NET supported platforms
author: jamesnk
description: Learn about the supported platforms for gRPC on .NET.
monikerRange: '>= aspnetcore-3.0'
ms.author: jamesnk
ms.date: 01/22/2021
no-loc: [appsettings.json, "ASP.NET Core Identity", cookie, Cookie, Blazor, "Blazor Server", "Blazor WebAssembly", "Identity", "Let's Encrypt", Razor, SignalR]
uid: grpc/supported-platforms
---
# gRPC for .NET supported platforms

By [James Newton-King](https://twitter.com/jamesnk)

## Device requirements

gRPC for .NET supports any device that .NET Core supports.

> [!div class="checklist"]
>
> * Windows
> * Linux
> * macOS&dagger;

&dagger;ASP.NET Core websites on macOS doesn't support HTTPS. gRPC clients on macOS can still use HTTPS when calling remote services.

## ASP.NET Core server requirements

gRPC services can be hosted on all built-in ASP.NET Core servers.

> [!div class="checklist"]
>
> * Kestrel
> * TestServer
> * IIS&dagger;
> * HttpSys&dagger;

> &dagger;IIS and HttpSys require .NET 5 and Windows XXXX or later

Reviewer: don't forget to update XXXX

Reviewer 2: Windows 10 Build 20241 or newer

@jkotalik Does IIS work with .NET 5

Author: Yes

Reviewer 3: 👍

## .NET version requirements

gRPC for .NET supports .NET Core 3 and .NET 5 or later.

> [!div class="checklist"]
>
> * .NET 5 or later
> * .NET Core 3

gRPC for .NET doesn't support running on .NET Framework and Xamarin/Mono. [gRPC C# core-library](https://grpc.io/docs/languages/csharp/quickstart/) is a third party library that supports .NET Framework and Xamarin/Mono. This library is not supported by Microsoft.

## Azure services

> [!div class="checklist"]
>
> * [Azure Kubernetes Service (AKS)](https://azure.microsoft.com/services/kubernetes-service/)
> * [Azure App Service](https://azure.microsoft.com/services/app-service/)&dagger;

> 

Reviewer 2: I haven't tried ACI, but the mega GitHub issue that had questions about it. It'd be good to add if it does indeed work

Author: Commenters in uber issue confirm it works.

> &dagger;Azure App Service doesn't support hosting gRPC over HTTP/2. gRPC-Web, which doesn't require HTTP/2 and supports most of gRPC's features, can be used instead.

Reviewer: ```suggestion
&dagger;Azure App Service doesn't support hosting gRPC over HTTP/2.  gRPC-Web doesn't require HTTP/2 and supports most of gRPC's features is the recommended alternative.
```

> &dagger;Azure App Service doesn't support hosting gRPC over HTTP/2.  gRPC-Web is the recommended alternative, it doesn't require HTTP/2 and supports most of gRPC's features.

Reviewer 2: I found this confusing since there is no prior mention of gRPC-web especially given that you've already mentioned IIS and HttpSys.

How about a separate H2 for gRPC-web that mentions the gRPC-web is an alternative protocol that is supported everywhere ASP.NET Core already runs

&Dagger;Blazor WebAssembly apps can call gRPC services with gRPC-Web.

## ASP.NET Core server requirements

gRPC services can be hosted on all built-in ASP.NET Core servers.

> [!div class="checklist"]
>
> * Kestrel
> * TestServer
> * IIS&dagger;
> * HttpSys&dagger;

&dagger;IIS and HttpSys require .NET 5 and Windows 10 Build 20241 or later.

## .NET version requirements

gRPC for .NET supports .NET Core 3 and .NET 5 or later.

> [!div class="checklist"]
>
> * .NET 5 or later
> * .NET Core 3

gRPC for .NET doesn't support running on .NET Framework and Xamarin. [gRPC C# core-library](https://grpc.io/docs/languages/csharp/quickstart/) is a third party library that supports .NET Framework and Xamarin. gRPC C-core is not supported by Microsoft.

## Azure services

> [!div class="checklist"]
>
> * [Azure Kubernetes Service (AKS)](https://azure.microsoft.com/services/kubernetes-service/)
> * [Azure App Service](https://azure.microsoft.com/services/app-service/)&dagger;

&dagger;Azure App Service doesn't support hosting gRPC over HTTP/2. gRPC-Web is a compatible alternative.

> Work is in-progress to improve support for gRPC with HTTP/2 in Azure App Service. For more information, see [this GitHub issue](https://github.com/dotnet/AspNetCore/issues/9020).

Author: Ok to say "work is in-progress"?
